### ISSUE TITLE
Vector tile layer crashes latest QGIS 3.22.3

### ISSUE THREAD

**Issue by nirvn** [qgis/QGIS#47315]
### What is the bug or the crash?

A vector tile layer is crashing QGIS 3.22.3 (but behaves find under QGIS master).

### Steps to reproduce the issue

1. On QGIS 3.22.{2,3}, load the following project:
[clip.zip](https://github.com/qgis/QGIS/files/8052613/clip.zip)
2. Notice QGIS dies :)

### Versions

Verified as crashing on 3.22.2 and 3.22.3, while it's fine on master.

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

Stack trace:

```
QgsSymbol::startRender :
QgsVectorTileBasicRenderer::renderTile :
QgsVectorTileMVTDecoder::operator= :
QObject::qt_static_metacall :
QgsVectorTileRendererData::tilePolygon :
QgsVectorTileMVTDecoder::operator= :
QObject::qt_static_metacall :
QgsTileDownloadManagerReply::requestFinished :
QgsTileDownloadManager::hasWorkerThreadRunning :
QMetaCallEvent::placeMetaCall :
QObject::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QCoreApplicationPrivate::sendPostedEvents :
QEventDispatcherWin32::processEvents :
QEventLoop::exec :
QgsVectorTileMVTDecoder::operator= :
QgsMapRendererParallelJob::renderLayerStatic :
QgsMapRendererParallelJob::renderingFinished :
QgsMapRendererParallelJob::renderingFinished :
QgsMapRendererParallelJob::takeLabelingResults :
QtConcurrent::ThreadEngineBase::run :
QThreadPoolPrivate::reset :
QThread::start :
BaseThreadInitThunk :
RtlUserThreadStart :
```

**Comment by nirvn** [comment]
A slight more useful stack trace:

```
Thread 0 Crashed:
0   libqfield_arm64-v8a.so          0x75a2766250        QgsSymbol::startRender (qgssymbol.cpp:492)
1   libqfield_arm64-v8a.so          0x75a352ea20        QgsVectorTileBasicRenderer::renderTile (qgsvectortilebasicrenderer.cpp:184)
2   libqfield_arm64-v8a.so          0x75a32c1328        QgsVectorTileLayerRenderer::decodeAndDrawTile (qgsvectortilelayerrenderer.cpp:254)
3   libQt5Core_arm64-v8a.so         0x75bcc2b3c8        doActivate<T>
4   libqfield_arm64-v8a.so          0x75a263d190        QgsVectorTileLoader::tileRequestFinished (moc_qgsvectortileloader.cpp:147)
5   libqfield_arm64-v8a.so          0x75a32c3554        QgsVectorTileLoader::tileReplyFinished (qgsvectortileloader.cpp:130)
6   libQt5Core_arm64-v8a.so         0x75bcc2b3c8        doActivate<T>
7   libqfield_arm64-v8a.so          0x75a2d78318        QtPrivate::FunctorCall<T>::call (qobjectdefs_impl.h:152)
8   libQt5Core_arm64-v8a.so         0x75bcc246b4        QObject::event
9   libQt5Widgets_arm64-v8a.so      0x75b439363c        QApplicationPrivate::notify_helper
```

Line 492 in 3.22.3 final is https://github.com/qgis/QGIS/blob/final-3_22_3/src/core/symbology/qgssymbol.cpp#L492 -- 

That looks to me like a null pointer being accessed, correct? If so, that could well be because the vector tile layer had its styles created in QGIS 3.23, which might result in some symbols not created when reading in 2.33. The code assume we are to be thrown a valid symbol pointer here all the time. A null protection check would help us prevent crashes.

**Comment by rouault** [comment]
Fixed per #47316

**Comment by nirvn** [comment]
@rouault , thanks, should I cleaned up my own fixed lamentation :wink:
